modules/dxp/apps/portal-workflow/portal-workflow-kaleo-designer-web/src/main/java/com/liferay/portal/workflow/kaleo/designer/web/internal/util/filter/KaleoDefinitionVersionVerifyVersionPredicate.java

```diff
@@ -0,0 +1,58 @@
+/**
+ * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
+ *
+ * The contents of this file are subject to the terms of the Liferay Enterprise
+ * Subscription License ("License"). You may not use this file except in
+ * compliance with the License. You can obtain a copy of the License by
+ * contacting Liferay, Inc. See the License for the specific language governing
+ * permissions and limitations under the License, including but not limited to
+ * distribution rights of the Software.
+ *
+ *
+ *
+ */
+
+package com.liferay.portal.workflow.kaleo.designer.web.internal.util.filter;
+
+import com.liferay.portal.kernel.exception.PortalException;
+import com.liferay.portal.kernel.log.Log;
+import com.liferay.portal.kernel.log.LogFactoryUtil;
+import com.liferay.portal.kernel.util.GetterUtil;
+import com.liferay.portal.workflow.kaleo.model.KaleoDefinition;
+import com.liferay.portal.workflow.kaleo.model.KaleoDefinitionVersion;
+
+import java.util.function.Predicate;
+
+/**
+ * @author Michael Cavalcanti
+ */
+public class KaleoDefinitionVersionVerifyVersionPredicate
+	implements Predicate<KaleoDefinitionVersion> {
+
+	@Override
+	public boolean test(KaleoDefinitionVersion kaleoDefinitionVersion) {
+		try {
+			KaleoDefinition kaleoDefinition =
+				kaleoDefinitionVersion.getKaleoDefinition();
+
+			if (kaleoDefinition.getVersion() == GetterUtil.getFloat(
```

Comment:
Hey @michaelmdelcavalcanti, what do you think about reversing the order of this condition?

Something like this:
```
if (GetterUtil.getFloat(kaleoDefinitionVersion.getVersion()) ==
		kaleoDefinition.getVersion()) {

	return true;
}
```